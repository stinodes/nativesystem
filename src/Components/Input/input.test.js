// @flow
import * as React from 'react'
import { Switch } from 'react-native'
import renderer from 'react-test-renderer'
import { Formik, Field } from 'formik'

import {
  TextInput,
  normalizeInputProps,
  formikFieldWrapper,
  createTheme,
} from '../index'

describe('Formik Helpers', () => {
  describe('mapInputProps', () => {
    test("maps the passed TextInput's props to a Formik-compatible shape", () => {
      const MappedTextInput = normalizeInputProps(
        {
          onChange: 'onChangeText',
        },
        TextInput,
      )
      const name = 'test input'
      const onChange = () => {}
      const onBlur = () => {}
      const onFocus = () => {}
      const tree = renderer.create(
        <MappedTextInput
          theme={createTheme()
            .useDefault()
            .done()}
          name={name}
          onChange={onChange}
          value=""
          onBlur={onBlur}
          onFocus={onFocus}
        />,
      )

      expect(tree.toJSON()).toMatchSnapshot()
    })
    test("defaults work with react-native's TextInput", () => {
      const MappedTextInput = normalizeInputProps({}, TextInput)
      const name = 'test input'
      const onChange = () => {}
      const onBlur = () => {}
      const onFocus = () => {}
      const tree = renderer.create(
        <MappedTextInput
          theme={createTheme()
            .useDefault()
            .done()}
          name={name}
          onChange={onChange}
          value=""
          onBlur={onBlur}
          onFocus={onFocus}
        />,
      )
      expect(tree.toJSON()).toMatchSnapshot()
    })
    test("maps the passed Switch's props to a Formik-compatible shape", () => {
      const MappedTextInput = normalizeInputProps(
        {
          onChange: 'onValueChange',
        },
        Switch,
      )
      const name = 'test input'
      const onChange = () => {}
      const onBlur = () => {}
      const onFocus = () => {}
      const tree = renderer.create(<Switch onChange={onChange} value={true} />)

      expect(tree.toJSON()).toMatchSnapshot()
    })
  })
  describe('FormikFieldWrapper', () => {
    test('wraps a compatible component to take a field and form prop', () => {
      const TextField = formikFieldWrapper(
        normalizeInputProps(
          {
            onChange: 'onChangeText',
          },
          TextInput,
        ),
      )
      const fieldProps = {
        name: 'textfield',
        value: '',
        onChange: () => {},
        onBlur: () => {},
        onFocus: () => {},
      }
      const tree = renderer.create(
        <TextField
          form={{}}
          field={fieldProps}
          theme={createTheme()
            .useDefault()
            .done()}
        />,
      )
      expect(tree.toJSON()).toMatchSnapshot()
    })
    test('works well with formik!', () => {
      const TextField = formikFieldWrapper(
        normalizeInputProps(
          {
            onChange: 'onChangeText',
          },
          TextInput,
        ),
      )
      const tree = renderer.create(
        <Formik initialValues={{ test: 'test' }}>
          {() => (
            <Field
              theme={createTheme()
                .useDefault()
                .done()}
              name="test"
              component={TextField}
            />
          )}
        </Formik>,
      )
      expect(tree.toJSON()).toMatchSnapshot()
    })
  })
})
