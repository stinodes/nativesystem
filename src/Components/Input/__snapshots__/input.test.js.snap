// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik Helpers FormikFieldWrapper works well with formik! 1`] = `
<TextInput
  allowFontScaling={true}
  form={
    Object {
      "dirty": false,
      "errors": Object {},
      "handleBlur": [Function],
      "handleChange": [Function],
      "handleReset": [Function],
      "handleSubmit": [Function],
      "initialValues": Object {
        "test": "test",
      },
      "isSubmitting": false,
      "isValid": false,
      "isValidating": false,
      "registerField": [Function],
      "resetForm": [Function],
      "setError": [Function],
      "setErrors": [Function],
      "setFieldError": [Function],
      "setFieldTouched": [Function],
      "setFieldValue": [Function],
      "setFormikState": [Function],
      "setStatus": [Function],
      "setSubmitting": [Function],
      "setTouched": [Function],
      "setValues": [Function],
      "submitCount": 0,
      "submitForm": [Function],
      "touched": Object {},
      "unregisterField": [Function],
      "validateField": [Function],
      "validateForm": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "values": Object {
        "test": "test",
      },
    }
  }
  name="test"
  onBlur={[Function]}
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 16.666666666666664,
      },
      Object {
        "color": "#2d3047",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="test"
/>
`;

exports[`Formik Helpers FormikFieldWrapper wraps a compatible component to take a field and form prop 1`] = `
<TextInput
  allowFontScaling={true}
  form={Object {}}
  name="textfield"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 16.666666666666664,
      },
      Object {
        "color": "#2d3047",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;

exports[`Formik Helpers mapInputProps defaults work with react-native's TextInput 1`] = `
<TextInput
  allowFontScaling={true}
  name="test input"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 16.666666666666664,
      },
      Object {
        "color": "#2d3047",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;

exports[`Formik Helpers mapInputProps maps the passed Switch's props to a Formik-compatible shape 1`] = `
<RCTSwitch
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={true}
/>
`;

exports[`Formik Helpers mapInputProps maps the passed TextInput's props to a Formik-compatible shape 1`] = `
<TextInput
  allowFontScaling={true}
  name="test input"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 16.666666666666664,
      },
      Object {
        "color": "#2d3047",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;

exports[`Formik Helpers mapInputProps maps the passed TextInput's props to a Formik-compatible shape 2`] = `
<RCTSwitch
  name="test input"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={true}
/>
`;

exports[`Formik Helpers mapInputProps maps the passed component's props to a Formik-compatible shape 1`] = `
<TextInput
  allowFontScaling={true}
  name="test input"
  onBlur={[Function]}
  onChangeText={[Function]}
  onFocus={[Function]}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 16.666666666666664,
      },
      Object {
        "color": "#2d3047",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
